src/software/gui/drawing/ssl_wrapper_packet.cpp

```diff
@@ -0,0 +1,54 @@
+#include "software/gui/drawing/ssl_wrapper_packet.h"
+
+#include "software/gui/drawing/ball.h"
+#include "software/gui/drawing/colors.h"
+#include "software/gui/drawing/field.h"
+#include "software/gui/drawing/robot.h"
+#include "software/proto/message_translation/ssl_detection.h"
+#include "software/proto/message_translation/ssl_geometry.h"
+
+void drawSSLWrapperPacket(QGraphicsScene* scene,
+                          const SSL_WrapperPacket& ssl_wrapper_packet)
+{
+    if (ssl_wrapper_packet.has_geometry())
+    {
+        auto geometry = ssl_wrapper_packet.geometry();
+        auto field    = createField(geometry);
+        if (field)
+        {
+            drawField(scene, field.value());
+        }
+    }
+
+    if (ssl_wrapper_packet.has_detection())
+    {
+        const auto detection = ssl_wrapper_packet.detection();
+
+        auto ball_detections = createBallDetections({detection});
+        for (const auto& ball : ball_detections)
+        {
+            drawBall(scene, ball);
+        }
```

Comment:
Since we could be drawing multiple balls, can we have a visual indication of confidence?

Reply:
Suggestions for what that might look like?

Reply:
brightness or saturation

Reply:
I should be able to do one of those. Meta-question, do we think that's useful? AFAIK "confidence" is a super vague value without any "real meaning" I know of, which is why it's not used in filtering currently (it didn't seem to help). It only seems potentially useful if we were drawing raw data and filtered data at the same time like you suggested, which implies running with SSL-Vision. This seems like a pretty rare case and I can't really think of why seeing a rough indicator of confidence is really useful (for debugging or something).

Reply:
Technically the filtered value we are "most confident" in will almost always cover the raw value underneath, so you wouldn't even necessarily see the raw confidence drawing.

Reply:
Hm, didn't know confidence was that useless. We don't need to do this then

Reply:
Yea it's surprisingly useless for something that sounds so useful